conjure-java-core/src/test/java/com/palantir/conjure/java/UndertowServiceEteTest.java

```diff
@@ -390,74 +375,73 @@ public void testOptionsOnGetIncludesHead() throws IOException {
     }
 
     @Test
-    public void testUnknownContentType() {
-        assertThatThrownBy(() -> {
-                    try {
-                        Futures.getUnchecked(binaryClient.postBinary(
-                                AuthHeader.valueOf("authHeader"),
-                                RequestBody.create(MediaType.parse("application/unsupported"), new byte[] {1, 2, 3})));
-                    } catch (UncheckedExecutionException e) {
-                        throw e.getCause();
-                    }
-                })
-                .isInstanceOf(RemoteException.class)
-                .hasMessageContaining("INVALID_ARGUMENT");
+    public void testUnknownContentType() throws Exception {
+        URL url = new URL("http://0.0.0.0:8080/test-example/api/binary");
+        HttpURLConnection con = (HttpURLConnection) url.openConnection();
+        con.setRequestMethod("POST");
+        con.setRequestProperty(HttpHeaders.CONTENT_TYPE, "application/unsupported");
+        con.setRequestProperty(
+                HttpHeaders.AUTHORIZATION, AuthHeader.valueOf("authHeader").toString());
+        con.setDoOutput(true);
+        con.getOutputStream().write(new byte[] {1, 2, 3});
+        assertThat(con.getResponseCode()).isEqualTo(415);
+        SerializableError serializableError =
+                CLIENT_OBJECT_MAPPER.readValue(con.getErrorStream(), SerializableError.class);
+        assertThat(serializableError.errorCode()).isEqualTo("INVALID_ARGUMENT");
     }
 
     @Test
     public void testSlashesInPathParam() {
         String expected = "foo/bar/baz/%2F";
         assertThat(client.path(AuthHeader.valueOf("bearer"), expected)).isEqualTo(expected);
-        assertThat(Futures.getUnchecked(retrofitClient.path(AuthHeader.valueOf("bearer"), expected)))
-                .isEqualTo(expected);
+        assertThat(client.path(AuthHeader.valueOf("bearer"), expected)).isEqualTo(expected);
     }
 
     @Test
     public void testPlusInPathParam() {
         String expected = "foo+bar";
         assertThat(client.path(AuthHeader.valueOf("bearer"), expected)).isEqualTo(expected);
-        assertThat(Futures.getUnchecked(retrofitClient.path(AuthHeader.valueOf("bearer"), expected)))
-                .isEqualTo(expected);
+        assertThat(client.path(AuthHeader.valueOf("bearer"), expected)).isEqualTo(expected);
     }
 
     @Test
     public void testSpaceInPathParam() {
         String expected = "foo bar";
         assertThat(client.path(AuthHeader.valueOf("bearer"), expected)).isEqualTo(expected);
-        assertThat(Futures.getUnchecked(retrofitClient.path(AuthHeader.valueOf("bearer"), expected)))
-                .isEqualTo(expected);
+        assertThat(client.path(AuthHeader.valueOf("bearer"), expected)).isEqualTo(expected);
```

Comment:
curious, what are the duplicate checks here? should one be testing the synchronous service and the other async service?

Reply:
Yea that is an oversight